python/cuspatial/cuspatial/io/geoseries_reader.py

```diff
@@ -0,0 +1,284 @@
+                # contains a buffer of sizes for each inner ring.
+                num_rings = 1
+                rings_current = offsets["polygons"]["rings"][-1]
+                offsets["polygons"]["rings"].append(
+                    len(geometry.exterior.coords) * 2 + rings_current
+                )
+                for interior in geometry.interiors:
+                    rings_current = offsets["polygons"]["rings"][-1]
+                    offsets["polygons"]["rings"].append(
+                        len(interior.coords) * 2 + rings_current
+                    )
+                    num_rings = num_rings + 1
+                current = offsets["polygons"]["polygons"][-1]
+                offsets["polygons"]["polygons"].append(num_rings + current)
+            elif isinstance(geometry, MultiPolygon):
+                current = offsets["polygons"]["polygons"][-1]
+                offsets["polygons"]["mpolys"].append(
+                    len(offsets["polygons"]["polygons"]) - 1
+                )
+                for poly in geometry:
+                    current = offsets["polygons"]["polygons"][-1]
+                    num_rings = 1
+                    rings_current = offsets["polygons"]["rings"][-1]
+                    offsets["polygons"]["rings"].append(
+                        len(poly.exterior.coords) * 2 + rings_current
+                    )
+                    for interior in poly.interiors:
+                        rings_current = offsets["polygons"]["rings"][-1]
+                        offsets["polygons"]["rings"].append(
+                            len(interior.coords) * 2 + rings_current
+                        )
+                        num_rings = num_rings + 1
+                    offsets["polygons"]["polygons"].append(num_rings + current)
+                offsets["polygons"]["mpolys"].append(
+                    len(offsets["polygons"]["polygons"]) - 1
+                )
+        return offsets
+
+    def _read_geometries(self, geoseries, offsets):
+        """
+        Creates a set of buffers sized to fit all of the geometries and
+        iteratively populates them with geometry coordinate values.
+
+        Parameters
+        ----------
+        geoseries : A GeoPandas GeoSeries object.
+        offsets : The set of offsets that correspond to the geoseries argument.
+        """
+        buffers = {
+            "points": cudf.Series(np.zeros(offsets["points"][-1])),
+            "multipoints": cudf.Series(np.zeros(offsets["multipoints"][-1])),
+            "lines": cudf.Series(np.zeros(offsets["lines"][-1])),
+            "polygons": {
+                "polygons": cudf.Series(
+                    np.zeros(len(offsets["polygons"]["polygons"]))
+                ),
+                "rings": cudf.Series(
+                    np.zeros(len(offsets["polygons"]["rings"]))
+                ),
+                "coords": cudf.Series(
+                    np.zeros(offsets["polygons"]["rings"][-1])
+                ),
```

Comment:
Shouldn't initialize GPU backed objects using numpy arrays. We have cudf column utilities for generating columns filled with zeros. If we need nd arrays here then we should use cupy instead of numpy.